textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -66,23 +63,41 @@ public void open() throws Exception {
         try {
             positionIndex = 0;
             attributeIndex = 0;
-            operator.open();
-            dictionaryValue = dictionary.getNextValue();
+            dictionaryValue = predicate.getNextDictionaryValue();
 
-            // Java regex is used to detect word boundaries for TextField match.
-            // '\b' is used to match the beginning and end of the word.
-            regex = "\\b" + dictionaryValue.toLowerCase() + "\\b";
-            pattern = Pattern.compile(regex);
+            if (predicate.getSourceOperatorType() == DataConstants.SourceOperatorType.PHRASEOPERATOR) {

```

Comment:
A general thought: instead of letting this DictionaryMatcher operator deal with different types of source operators explicitly, can we push the differences into the source operator to make the logic of this operator independent of the source?